game.cpp

```diff
@@ -352,6 +361,30 @@ void game::eat_food(food& f)
   f.reset_timer();
 }
 
+void game::increment_shoot_cool_down_timers()
+{
+  for (player &p : m_player)
+    {
+      if (p.is_shoot_cool_down())
+      {
+        p.increment_shoot_cool_down_timer();
+      }
+    }
+}
+
+void game::reset_shoot_cool_down_status()
+{
+  for (player &p : m_player)
+    {
+      float player_shoot_fire_rate = projectile::m_fire_rate / p.get_shoot_fire_rate_multiplier();
```

Comment:
I don't understand why the fire rate should be player-specific here... My guess is that you have future plans for mechanisms that will allow players to shoot more or less fast after eating some food, going in a specific part of the terrain or something else like this? If so, this is great and I would encourage you to do it, but for now this adds complexity that is unnecessary to solve the issue (#513).
For now, I would advise to keep this idea aside, and if you have a greater plan for changing the fire rate based on a player's condition, write spearate issues and tests for it first 😉

Reply:
I do plan to implement a weapon upgrade system in the near future, but I will surely break down the whole thing into several issues and solve them one at a time.

Reply:
And one thing that makes the code less robust here is that I divided the fire rate by the player-specific multiplier, this will probably generate a float, which may introduce issues in later development since ticks should be integer. If I implement player-specific fire rate again, I may need to round the value, or use addition instead of multiplication or division? I think this would be interesting if we have time to discuss the game design in a later meeting.

Reply:
@EvoLandEco in the future, there will be a `player_options` class, containing all constants a player has, such as, for example, the firing rate. I feel these can easily differ per player :+1: